gensim/models/keyedvectors.py

```diff
@@ -1062,7 +1062,7 @@ def evaluate_word_analogies(self, analogies, restrict_vocab=300000, case_insensi
         logger.info("Evaluating word analogies for top %i words in the model on %s", restrict_vocab, analogies)
         sections, section = [], None
         quadruplets_no = 0
-        for line_no, line in enumerate(utils.smart_open(analogies)):
+        for line_no, line in enumerate(utils.open(analogies, 'rb')):
```

Comment:
Context manager

Reply:
done, changed almost everywhere (except 2 places that handle the closing explicitly)